### PR TITLE
Make `workflow_params` optional for WES workflow submissions

### LINKED ISSUE #4037: Toil WES support change workflow_params to be optional
When launching a workflow using a Toil WES server, the `workflow_params` (input JSON file) is required, even if the workflow itself takes no inputs. Not including the `workflow_params` body part in the HTTP request causes the WEs server to return an error:
```
io.openapi.wes.client.ApiException: {
  "msg": "Missing 'workflow_params' in submission",
  "status_code": 400
}
```

Ideally, this parameter should be optional, to allow easy launching of input parameter-free workflows. This is pretty easy to work around though, as passing in a dummy test parameter file suppresses the exception.

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/TOIL-1143)
┆friendlyId: TOIL-1143

### PR DESCRIPTION
**Pull request by adamnovak**
This should fix #4037 by allowing WES requests to omit `workflow_params` from the POST data. If that happens, we will now treat it the same as if they had passed an empty JSON-serialized object.

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * Toil WES server now accepts requests that leave out `workflow_params`.

## Reviewer Checklist

 * [ ] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [ ] Read through the code changes. Make sure that it doesn't have:
    * [ ] Addition of trailing whitespace.
    * [ ] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [ ] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [ ] New functions or classes without informative docstrings.
    * [ ] Changes to semantics not reflected in the relevant docstrings.
    * [ ] New or changed command line options for Toil workflows that are not reflected in `docs/running/{cliOptions,cwl,wdl}.rst` 
    * [ ] New features without tests.
* [ ] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

File: src/toil/server/wes/abstract_backend.py [hunk 1]
```diff
@@ -256,7 +256,8 @@ def collect_attachments(self, run_id: Optional[str], temp_dir: Optional[str]) ->
         else:
             raise MalformedRequestException("Missing 'workflow_url' in submission")
 
-        if "workflow_params" not in body:
-            raise MalformedRequestException("Missing 'workflow_params' in submission")
+        if "workflow_params" in body and not isinstance(body["workflow_params"], dict):
```
Comment (DailyDreaming) [inline src/toil/server/wes/abstract_backend.py:259]: Suggestion:  Could be shortened to:

```
if not isinstance(body.get("workflow_params"), dict):
```

**Review by DailyDreaming — APPROVED** [review]
LGTM.  One minor suggestion.

File: src/toil/server/wes/abstract_backend.py (hunk 1 above)
Comment (adamnovak) [inline src/toil/server/wes/abstract_backend.py:259]:
```suggestion
        if not isinstance(body.get("workflow_params"), dict):
            # get will return None if the key is absent, which isn't a dict
```

Comment (adamnovak) [reply src/toil/server/wes/abstract_backend.py]: This doesn't actually work. If it's not there, we will `get` `None`, which isn't a `dict`, so we'll pass the `if`. We don't actually want to run the body when `workflow_params` isn't there.

**Pull request merged**